<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>The RforMassSpectrometry project and meta-package • RforMassSpectrometry</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="The RforMassSpectrometry project and meta-package">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">RforMassSpectrometry</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/RforMassSpectrometry.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>The RforMassSpectrometry project and meta-package</h1>
                        <h4 class="author">Laurent Gatto</h4>
            <address class="author_afil">
      de Duve Institute, UCLouvain, Belgium<br><h4 class="author">Sebastian Gibb</h4>
            <address class="author_afil">
      University Medicine Greifswald, Germany<br><h4 class="author">Johannes Rainer</h4>
            <address class="author_afil">
      Eurac Research, Bolzano, Italy<br><div class="hidden name"><code>RforMassSpectrometry.Rmd</code></div>

    </address>
</address>
</address>
</div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p><code>RforMassSpectrometry</code> is a meta-package, to manage and document the RforMassSpectrometry (<a href="https://www.rformassspectrometry.org/" class="uri">https://www.rformassspectrometry.org/</a>) project. The goal of the project to provide efficient, thoroughly documented, tested and flexible R software for the analysis and interpretation of high throughput mass spectrometry assays, including proteomics and metabolomics experiments. The project formalises the longtime collaborative development efforts of its members under the RforMassSpectrometry organisation to facilitate dissemination and accessibility of their work.</p>
</div>
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<p>To install the core R for Mass Spectrometry packages, you will need the <code>BiocManager</code> package, available on CRAN. If not already available, install it with</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="cf">if</span> (<span class="op">!</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">require</a></span>(<span class="st">"BiocManager"</span>))</a>
<a class="sourceLine" id="cb1-2" data-line-number="2">    <span class="kw">install.package</span>(<span class="st">"BiocManager"</span>)</a></code></pre></div>
<p>To proceed with the installation, run</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">BiocManager<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/BiocManager/topics/install">install</a></span>(<span class="st">"RforMassSpectrometry/RforMassSpectrometry"</span>)</a></code></pre></div>
<p>Note that some very recent Bioconductor will be needed to install and run the suite of R for Mass Spectrometry packages, and will require a recent version of R and possibly even installing the <a href="http://bioconductor.org/developers/how-to/useDevel/">Bioconductor development version</a>.</p>
<p>Loading the <code>RforMassSpectrometry</code> package with</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(<span class="st">"RforMassSpectrometry"</span>)</a></code></pre></div>
<pre><code>## 
## This is RforMassSpectrometry version 0.0.1 
##   Please not that it is a very early stage project.
##   Visit https://RforMassSpectrometry.org to get started.</code></pre>
<p>will load the required core packages.</p>
</div>
<div id="code-of-conduct" class="section level1">
<h1 class="hasAnchor">
<a href="#code-of-conduct" class="anchor"></a>Code of Conduct</h1>
<p>As contributors and maintainers of the RforMassSpectrometry initiative, we pledge to respect all people who contribute through reporting issues, posting feature requests, updating documentation, submitting pull requests or patches, and other activities. We are committed to making participation in this project a harassment-free experience for everyone, regardless of level of experience, gender, gender identity and expression, sexual orientation, disability, personal appearance, body size, race, ethnicity, age, or religion. Examples of unacceptable behaviour by participants include the use of sexual language or imagery, derogatory comments or personal attacks, trolling, public or private harassment, insults, or other unprofessional conduct. Project maintainers have the right and responsibility to remove, edit, or reject comments, commits, code, wiki edits, issues, and other contributions that are not aligned to this Code of Conduct. Project maintainers who do not follow the Code of Conduct may be removed from the project team. Instances of abusive, harassing, or otherwise unacceptable behaviour may be reported by opening an issue or contacting one or more of the project maintainers.</p>
</div>
<div id="contributions" class="section level1">
<h1 class="hasAnchor">
<a href="#contributions" class="anchor"></a>Contributions</h1>
<p>The RforMassSpectrometry project welcomes contributions in the form of ideas, documentation, code, packages, fixes (bugs and typos), … We invite you to discuss any of these opening a GitHub issue, either in the <a href="https://github.com/rformassspectrometry/RforMassSpectrometry/issues"><code>RforMassSpectrometry</code> package</a>, or any other package of relevant.</p>
<p>If you have any direct contributions to an existing repository, we suggest to first discuss these as an issue if it implies any substantial changes (no need for an issue if its a simple typo, and we really appreciate those). Contributions, even those from core members, are always expected to come in the form of a pull request (PR) that then undergoes a formal review. To do this, start by forking the package you want to contribute to. It is also advised to work in, and make the PR, from a non-master branch. Good branch names include <code>feat-X</code>, <code>bug-Y</code>, <code>docs-Z</code> for new features, bug fixes and documentation updates respectively. The reviewers might ask for additional clarifications and/or changes prior to merging the PR.</p>
<p>When contemplating contributing code, please follow the style guide below.</p>
<p>We also anticipate to open up some specific requests for contributions and invite specific members of the community, if we identify a specific need and/or a specific developer to address that need.</p>
<p>Reviewers we will do their best to help you to get the PR merged<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>.</p>
<div id="credit-where-credit-is-due" class="section level3 unnumbered">
<h3 class="hasAnchor">
<a href="#credit-where-credit-is-due" class="anchor"></a>Credit where credit is due</h3>
<p>We also want to offer proper credit to those that provide substantial contributions. Substantial contributions include, for example</p>
<ul>
<li>new functionality that goes beyond e.g. a simple plot function</li>
<li>bug fixes combined with related unit tests</li>
<li>extended documentation, examples or vignettes describing a complete (potentially alternative) analysis workflow or use-case</li>
</ul>
<p>Fixing typo(s) isn’t considered a substantial contribution per-se, even though they are highly appreciated.</p>
<p>Once a substantial contribution passes review and is merged, we will add the contributor in the package’s author list as an official contributor (<code>ctb</code> role) or even author (<code>aut</code> role) for more substantial contributions. The contributors name will also be added to the RforMassSpectrometry web page and to any relevant publications that directly stem from their work.</p>
</div>
</div>
<div id="coding-style" class="section level1">
<h1 class="hasAnchor">
<a href="#coding-style" class="anchor"></a>Coding style</h1>
<p>The RforMassSpectromery packages follows the <a href="https://bioconductor.org/developers/how-to/coding-style/">Bioconductor style guide</a>. In particular</p>
<ul>
<li>Do not use <code>.</code> when naming symbols.</li>
<li>A leading <code>.</code> can be used for hidden/local functions or variables.</li>
<li>Snake case should be restricted to internal functions. For consistency, we favour camel case for public functions.</li>
<li>Class names should start with a capital and each class should posses a constructor with identical name. Running the constructor without any input should produce a valid empty object.</li>
<li>Use <code>##</code> to start full-line comments.</li>
<li>For roxygen headers <code>#'</code> is preferred, although <code>##'</code> is tolerated.</li>
<li>Use spaces between <code>=</code> in function arguments or class definition: <code>f(a = 1, b = 2)</code>.</li>
<li>Always use a space after a comma: <code>a, b, c</code>.</li>
<li>Always use spaces around binary operators: <code>a + b</code>.</li>
<li>Lines should be kept shorter than 80 characters. For example the following code isn’t accepted</li>
</ul>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="co"># no wrap at 80</span></a>
<a class="sourceLine" id="cb5-2" data-line-number="2">someVeryLongVariableName &lt;-<span class="st"> </span><span class="kw">someVeryLongFunctionName</span>(<span class="dt">withSomeEvenLongerFunctionArgumentA =</span> <span class="dv">1</span>, <span class="dt">withSomeEvenLongerFunctionArgumentB =</span> <span class="dv">2</span>)</a></code></pre></div>
<p>and should be wrapped as shown below:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="co"># alternative 1</span></a>
<a class="sourceLine" id="cb6-2" data-line-number="2">someVeryLongVariableName &lt;-</a>
<a class="sourceLine" id="cb6-3" data-line-number="3"><span class="st">    </span><span class="kw">someVeryLongFunctionName</span>(<span class="dt">withSomeEvenLongerFunctionArgumentA =</span> <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb6-4" data-line-number="4">                             <span class="dt">withSomeEvenLongerFunctionArgumentB =</span> <span class="dv">2</span>)</a>
<a class="sourceLine" id="cb6-5" data-line-number="5"><span class="co"># alternative 2</span></a>
<a class="sourceLine" id="cb6-6" data-line-number="6">someVeryLongVariableName &lt;-<span class="st"> </span><span class="kw">someVeryLongFunctionName</span>(</a>
<a class="sourceLine" id="cb6-7" data-line-number="7">    <span class="dt">withSomeEvenLongerFunctionArgumentA =</span> <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb6-8" data-line-number="8">    <span class="dt">withSomeEvenLongerFunctionArgumentB =</span> <span class="dv">2</span>)</a></code></pre></div>
</div>
<div id="contact" class="section level1">
<h1 class="hasAnchor">
<a href="#contact" class="anchor"></a>Contact</h1>
<p>Please see the <a href="https://www.rformassspectrometry.org/contact/">contact page</a> for instructions on how to get in touch.</p>
</div>
<div id="session-info" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<pre><code>## R version 3.6.0 (2017-01-27)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 14.04.5 LTS
## 
## Matrix products: default
## BLAS:   /home/travis/R-bin/lib/R/lib/libRblas.so
## LAPACK: /home/travis/R-bin/lib/R/lib/libRlapack.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] RforMassSpectrometry_0.0.1 BiocStyle_2.13.2          
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_1.0.1                  compiler_3.6.0             
##  [3] BiocManager_1.30.4          GenomeInfoDb_1.21.1        
##  [5] XVector_0.25.0              ProtGenerics_1.17.2        
##  [7] zlibbioc_1.31.0             Features_0.1.0             
##  [9] bitops_1.0-6                tools_3.6.0                
## [11] digest_0.6.20               lattice_0.20-38            
## [13] evaluate_0.14               memoise_1.1.0              
## [15] rlang_0.4.0                 Matrix_1.2-17              
## [17] DelayedArray_0.11.4         commonmark_1.7             
## [19] yaml_2.2.0                  parallel_3.6.0             
## [21] pkgdown_1.3.0               xfun_0.8                   
## [23] Spectra_0.2.0               GenomeInfoDbData_1.2.1     
## [25] stringr_1.4.0               roxygen2_6.1.1             
## [27] xml2_1.2.0                  knitr_1.23                 
## [29] desc_1.2.0                  fs_1.3.1                   
## [31] S4Vectors_0.23.17           IRanges_2.19.10            
## [33] grid_3.6.0                  MultiAssayExperiment_1.11.4
## [35] stats4_3.6.0                rprojroot_1.3-2            
## [37] Biobase_2.45.0              R6_2.4.0                   
## [39] BiocParallel_1.19.0         rmarkdown_1.14             
## [41] bookdown_0.12               AnnotationFilter_1.9.0     
## [43] magrittr_1.5                matrixStats_0.54.0         
## [45] backports_1.1.4             htmltools_0.3.6            
## [47] MASS_7.3-51.4               GenomicRanges_1.37.14      
## [49] BiocGenerics_0.31.5         MsExperiment_0.0.1         
## [51] SummarizedExperiment_1.15.5 assertthat_0.2.1           
## [53] stringi_1.4.3               lazyeval_0.2.2             
## [55] RCurl_1.95-4.12             crayon_1.3.4</code></pre>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p>Useful reading: The <a href="https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#reviewing-pull-requests">nodejs reviewing PRs for reviewers</a> guide<a href="#fnref1" class="footnote-back">↩</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction">Introduction</a></li>
      <li><a href="#installation">Installation</a></li>
      <li><a href="#code-of-conduct">Code of Conduct</a></li>
      <li><a href="#contributions">Contributions</a></li>
      <li><a href="#coding-style">Coding style</a></li>
      <li><a href="#contact">Contact</a></li>
      <li><a href="#session-info">Session info</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Laurent Gatto, Johannes Rainer, Sebastian Gibb.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
